File: Python3/2145. Count the Hidden Sequences.py
class Solution:
    def numberOfArrays(self, differences: List[int], lower: int, upper: int) -> int:
        min_diff, max_diff, cur = 0, 0, 0

        for d in differences:
            cur += d
            min_diff = min(min_diff, cur)
            max_diff = max(max_diff, cur)
            if max_diff - min_diff > upper - lower:
                return 0

        return upper - lower - (max_diff - min_diff) + 1
        

File: Java/2145. Count the Hidden Sequences.java
class Solution {
    public int numberOfArrays(int[] differences, int lower, int upper) {
        int max_diff = 0, min_diff = 0, cur = 0;
        for (int d: differences) {
            cur += d;
            max_diff = Math.max(max_diff, cur);
            min_diff = Math.min(min_diff, cur);
            if (max_diff - min_diff > upper - lower) {
                return 0;
            }
        }
        return upper - lower - max_diff + min_diff + 1;
    }
}